### ISSUE TITLE
.bam extension required for -b option to methylartist regions

### ISSUE THREAD

**Issue by lfearnley** [adamewing/methylartist#25]
Hi,

Not sure if this is a bug or a documentation issue - it could be handled either way. I'm currently storing BAMs containing modified basecalls as files with the extension .modbam to differentiate them from those with non-modified basecalls.

The `-b` flag in `methylartist regions` has a different behaviour dependent on the file extension - if I run it on the .modbam file, I get:

```2022-08-31 08:50:04,401 assuming test_sample.guppy5mc.haplotagged.modbam contains a list of .bams
Traceback (most recent call last):
  File "/stornext/HPCScratch/home/fearnley.l/miniconda3/envs/megalodon/bin/methylartist", line 4855, in <module>
    main(args)
  File "/stornext/HPCScratch/home/fearnley.l/miniconda3/envs/megalodon/bin/methylartist", line 4570, in main
    args.func(args)
  File "/stornext/HPCScratch/home/fearnley.l/miniconda3/envs/megalodon/bin/methylartist", line 3476, in region
    for line in bam_list:
  File "/stornext/HPCScratch/home/fearnley.l/miniconda3/envs/megalodon/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

And if I symlink the same file to have a .bam extension with (`ln -s test_sample.guppy5mc.haplotagged.modbam test_sample.guppy5mc.haplotagged.modbam.bam`), updating its index filename appropriately, methylartist works.

It looks like this comes down to a file extension check that determines whether a file is passed to pysam or not (e.g. lines 2096-2104).

**Comment by adamewing** [comment]
Ah, yes, thank you. I'll try to catch this and output something more helpful and/or add a flag to force `.bam` input.

**Comment by adamewing** [comment]
fixed in 1.2.5
